etc/rc.d/arp_respond.lua

```diff
@@ -0,0 +1,24 @@
+local component = require("component")
+local thread = require("thread")
+local event = require("event")
+require "file"
+
+local function filterARP(type, dest, origin, port, _, protocol, ip_address)
+    -- check if the request is an arp request and if it is on the right port and if it is directed to this device
+    if type ~= "modem_message" or port ~= 1 or protocol ~= "ARP_DISCOVER" or ip_address ~= read_config_key("/etc/network/interface", "ip") then
+        return false
+    end
+    return true
+end
+
+function start()
```

Comment:
Yes but this would only work properly if the the process is restarted whenever networking is re-enabled. In short we should have one process that initializes all background processes like in real linux

Reply:
In that case we should not depend on ports to check if networking is enabled.

Reply:
Also checking if port 10 is open would create a dependency that we maybe not always want.

Reply:
Now for the stop part i will keep the "while true do" because the thread will almost all of the time be in `event.pullFiltered(filterARP)`. If we close the port the thread will stay on this line forever because it is listening for events on a port that is closed. Even if it could still receive a message we don't want it to respond anyway. So the most practical way of dealing with this is to simply kill the thread and close the port.